modules/gapi/include/opencv2/gapi/gtype_traits.hpp

```diff
@@ -47,6 +48,11 @@ namespace detail
         static constexpr const ArgKind kind = ArgKind::GMAT;
         static constexpr const GShape shape = GShape::GMAT;
     };
+    template<>           struct GTypeTraits<cv::GMatP>
+    {
+        static constexpr const ArgKind kind = ArgKind::GMATP;
```

Comment:
Why does it need a special Kind enum?

Reply:
It's required by GArg->GProtoArg conversion